src/Bounds.cpp

```diff
@@ -1029,7 +1029,14 @@ class Bounds : public IRVisitor {
                 // For some of these intrinsics applied to integer
                 // types we can go a little further.
                 if (t.is_int() || t.is_uint()) {
-                    if (op->is_intrinsic(Call::shift_right)) {
```

Comment:
The whole point of the change was to remove the use of the simplifier here. Does the other PR (#3642) solve your problem? If not we should canonicalize without the simplifier to avoid a recursive descent into the LHS.